src/vue/common/Pagination.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="pagination">
+    <slot />
+    <button
+      class="pagination-btn"
+      @click="$emit(PAGINATION_EVENTS.loadNextPage)">
+      {{ 'pagination.btn-label-more' | globalize }}
+    </button>
+  </div>
+</template>
+
+<script>
+import { ErrorHandler } from '@/js/helpers/error-handler'
+const PAGINATION_EVENTS = {
+  loadFirstPage: 'load-first-page',
+  loadNextPage: 'load-next-page'
+}
+
+export default {
+  name: '',
+  components: {},
+  filters: {},
+  props: {
+    firstPageLoader: {
+      type: Function,
+      require: true,
+      default: () => function () {}
```

Comment:
1 - Why do you need the return value to be a function? that doesn't make any sense
 2 - Why do you need the default value for the required property?